docs/source/sysadmin_guide/validator_key_permissioning.rst

```diff
@@ -0,0 +1,336 @@
+    }
+
+  The requester then signs the payload message and returns a response that
+  includes the following:
+
+  .. code-block:: protobuf
+
+    message AuthorizationChallengeSubmit {
+      // public key of node
+      string public_key = 1;
+
+      // payload bytes that was signed to make the signature
+      bytes payload = 2;
+
+      // signature derived from signing the challenge payload
+      string signature = 3;
+
+      // A set of requested Roles
+      repeated RoleType roles = 4;
+    }
+
+  The requester may also request ALL. The validator will respond with a
+  status that says whether the challenge was accepted and the roles that the
+  connection is allowed take on.
+
+  .. code-block:: protobuf
+
+    message AuthorizationChallengeResult {
+      // The approved roles for that connection
+      repeated RoleType roles = 1;
+    }
+
+  Message flow for Challenge Authorization:
+
+  .. image:: ../images/challenge_authorization.*
+     :width: 80%
+     :align: center
+     :alt: Trust Authorization Flow
+
+  When the validator receives an AuthorizationChallengeSubmit message, it
+  verifies the public key against the signature. If the public key is verified,
+  the requested roles is checked against the stored roles to see if the
+  public key is included in the policy. If the node’s response is accepted, the
+  node’s public key is stored and the requester may start sending messages
+  for the approved roles.
+
+  If the requester wanted a role that is either not available on the endpoint
+  or the requester does not have access to one of the roles requested, the
+  challenge will be rejected and the connection is closed. At that point
+  the requester will need to restart the connection process.
+
+Authorization Violation
+  If at any time a requester tries to send a message that is against its
+  allowed permission, the validator responds with an AuthorizationViolation
+  message and the connection is closed. If that requester wishes to rejoin
+  the network, it will need to go back through the connection and authorization
+  process described above.
+
+  .. code-block:: protobuf
+
+    message AuthorizationViolation {
+      // The Role the requester did not have access to
+      RoleType violation = 1;
+    }
+
+Network Roles
+-------------
+To configure on-chain roles, the signer of identity transactions needs to have
+their public key set in the Setting "sawtooth.identity.allowed_keys". Only
+those transactors whose public keys are in that setting are allowed to update
+roles and policies.
+
+.. code-block:: console
+
+  $ sawtooth config proposal create sawtooth.identity.allowed_keys=02b2be336a6ada8f96881cd55fd848c10386d99d0a05e1778d2fc1c60c2783c2f4
+
+Once your signer key is stored in the setting, the identity cli can be used to
+set and update roles and policies. Make sure that the identity transaction
+processor and the the rest api are running.
+
+.. code-block:: console
+
+  $ sawtooth identity policy create -h
```

Comment:
Rather than commit the usage here, these should be added to the CLI Reference section and have their usage generated automatically. A link to that page would go here then.